core/src/executor/select/mod.rs

```diff
@@ -166,40 +171,80 @@ pub async fn select_with_labels<'a>(
-    let table = Table::new(relation)?;
+    let (rows, columns) = match relation {
+        TableFactor::Table { .. } => {
+            let table = Table::new(relation)?;
+            let columns = fetch_columns(storage, table.get_name()).await?;
+            let columns = Rc::from(columns);
+            (
+                Rows::Table(fetch_blended(storage, table, Rc::clone(&columns)).await?),
+                columns,
+            )
+        }
+        TableFactor::Derived { subquery, alias } => {
+            let (labels, inline_view) = select_with_labels(storage, subquery, None, true).await?;
+            let inline_view = inline_view.try_collect::<Vec<_>>().await?;
+            let labels_rc = Rc::from(labels.to_owned());
+            let labels = Rc::clone(&labels_rc);
+            let rows = inline_view.into_iter().map(move |row| {
+                Ok(BlendContext::new(
+                    &alias.name,
+                    Rc::clone(&labels_rc),
+                    Some(row),
+                    None,
+                ))
+            });
+            (Rows::Derived(stream::iter(rows)), Rc::clone(&labels))
+        }
+    };
 
-    let columns = fetch_columns(storage, table.get_name()).await?;
     let join_columns = stream::iter(joins.iter())
         .map(Ok::<_, Error>)
-        .and_then(|join| {
-            let table = Table::new(&join.relation);
-
-            async move {
-                let table = table?;
-                let table_alias = table.get_alias();
-                let table_name = table.get_name();
-
-                let columns = fetch_columns(storage, table_name).await?;
-
-                Ok((table_alias, columns))
+        .and_then(|join| async move {
+            match &join.relation {
+                TableFactor::Table { .. } => {
+                    let table_alias = join.relation.get_alias()?;
+                    let table_name = join.relation.get_name()?;
+                    let columns = fetch_columns(storage, table_name).await?;
+
+                    Ok((table_alias, columns))
+                }
+                TableFactor::Derived {
+                    subquery:
+                        Query {
+                            body: SetExpr::Select(statement),
+                            ..
+                        },
+                    alias: _,
+                } => {
+                    let Select {
+                        from: TableWithJoins { relation, .. },
+                        ..
+                    } = statement.as_ref();
+
+                    let Select { projection, .. } = statement.as_ref();
+                    let inner_table_name = relation.get_name()?;
+                    let columns = fetch_columns(storage, inner_table_name).await?;
+                    let join_columns = &[(&"null".to_string(), vec![])]; // todo: join_columns should be Option?
+                    let columns = get_labels(projection, inner_table_name, &columns, join_columns)?;
```

Comment:
To get join_columns, I used get_labels but it needs join_columns.
Should join_columns be Option?

Reply:
I think.. as you've successfully add a new fetch method of `fetch_relation` to fetch rows using `TableFactor` param.

Here this also looks possible to find the common pattern in this module.
In this code, we may need to add `fetch_columns` for `TableFactor`.

There exists two cases which we handle `TableFactor` enum directly in this file `select/mod.rs`.
fetching columns using TableFactor might be a key clue.. I think.

I will also look and dig more on this and share if I find something.